core/src/main/java/org/apache/calcite/sql/SqlCreate.java

```diff
@@ -26,30 +26,30 @@
 public abstract class SqlCreate extends SqlDdl {
 
   /** Whether "OR REPLACE" was specified. */
-  boolean replace;
+  final SqlCreateOrReplace replace;
 
   /** Whether "IF NOT EXISTS" was specified. */
   public final boolean ifNotExists;
 
   /** Creates a SqlCreate. */
-  public SqlCreate(SqlOperator operator, SqlParserPos pos, boolean replace,
+  public SqlCreate(SqlOperator operator, SqlParserPos pos, SqlCreateOrReplace replace,
       boolean ifNotExists) {
     super(operator, pos);
     this.replace = replace;
     this.ifNotExists = ifNotExists;
   }
 
   @Deprecated // to be removed before 2.0
-  public SqlCreate(SqlParserPos pos, boolean replace) {
+  public SqlCreate(SqlParserPos pos, SqlCreateOrReplace replace) {
```

Comment:
When I came across the "@Deprecated" line, I came to realization that this might be a well documented component that the **Adapters** in Calcite project uses. Could you clarify in the design doc how this change won't break codes that depends on this functionality?

Reply:
I'm a little confused - all the parameter types have been updated throughout the project (boolean -> enum), and all the tests are passing. The design doc mentions making this boolean->enum change in all dependencies (I felt there were too many small search-and-replace changes throughout the project for it to be worth listing them all out). Could you clarify why you think this change in particular needs more documentation?

Reply:
I am okay with it since no test fails because of the change.